src/index.js

```diff
@@ -119,8 +120,23 @@ function(err, results) {
     shell.exec('npm version --no-git-tag-version ' + oldVersion);
   }
 
-  // ### STEP 8 - Run if any pre commit script has been specified (DESTRUCTIVE OPERATION)
-  lib.runPreCommitScript(program.preCommit, version);
+  // ### STEP 8 - Run if any pre commit script has been specified (DESTRUCTIVE OPERATION?)
+  if (program.dryRun) {
+    log.info('>>> Skipping pre-commit script');
+  } else {
+    lib.runScript(program.preCommit, 'pre-commit', version);
+  }
+
   // ### STEP 9 - Tag and push (DESTRUCTIVE OPERATION)
-  if (!program.dryrun) lib.addFilesAndCreateTag(version);
+  if (!program.dryrun) {
+    let pushResultCode = lib.addFilesAndCreateTag(version);
+
+    // ### STEP 10 - Run after successful push (DESTRUCTIVE OPERATION)
+    if (pushResultCode === 0) {
```

Comment:
It looks good here now. I think it is good to keep the dryrun non-destructive. :+1: